src/react-components/ory/user-logout-card.spec.tsx

```diff
@@ -0,0 +1,41 @@
+import { expect, test } from "@playwright/experimental-ct-react"
+import { Locator } from "@playwright/test"
+import { UserLogoutCard } from "./user-logout-card"
+
+let component: Locator
+
+test.beforeEach(async ({ mount }) => {
+  component = await mount(
+    <UserLogoutCard
+      csrfToken="CSRF token"
+      action="https://example.com/logout"
+      challenge="logout challenge"
+    />,
+  )
+})
+
+test("submits when user clicks 'Yes'", async () => {
+  const [request] = await Promise.all([
+    component.page().waitForRequest("https://example.com/logout"),
+    component.locator("button", { hasText: /yes/i }).click(),
+  ])
+
+  expect(request.method()).toBe("POST")
+  const data = request.postDataJSON()
+  expect(data).toHaveProperty("_csrf", "CSRF token")
+  expect(data).toHaveProperty("challenge", "logout challenge")
+  expect(data).toHaveProperty("submit", "Yes")
+})
+
+test("submits when user clicks 'No'", async () => {
+  const [request] = await Promise.all([
+    component.page().waitForRequest("https://example.com/logout"),
+    component.locator("button", { hasText: /no/i }).click(),
+  ])
```

Comment:
This can be racy (see [here](https://www.notion.so/Playwright-E2E-Testing-Ory-Network-a88cba00aa3c4e3982bfa302e78be006?pvs=4#1f7b786a20ca48fb8c432c59f267cc53) and [here](https://playwright.dev/docs/api/class-page#page-wait-for-request))
You should first create the promise, then trigger the action and then await the promise.
```suggestion
  const requestPromise = component.page().waitForRequest("https://example.com/logout")
  await component.locator("button", { hasText: /no/i }).click()
  const request = await requestPromise
```